src/Styleguide/Elements/__tests__/RadioGroup.test.tsx

```diff
@@ -1,64 +1,144 @@
     const spy = jest.fn()
-    const wrapper = mount(<RadioGroup onSelect={spy} options={options} />)
+    const wrapper = mount(
+      <RadioGroup onSelect={spy}>
+        <Radio value="SHIP">Provide shipping address</Radio>
+        <Radio value="PICKUP">Arrange for pickup</Radio>
+      </RadioGroup>
+    )
+
     expect(wrapper.text()).toContain("Provide shipping address")
     expect(wrapper.text()).toContain("Arrange for pickup")
-    expect(wrapper.find("StyledRadio").length).toBe(2)
+
     wrapper
-      .find("StyledRadio")
+      .find("Radio")
       .first()
       .simulate("click")
+
     expect(spy).toHaveBeenCalled()
   })
 
   it("renders default value on mount", () => {
     const wrapper = mount(
-      <RadioGroup options={options} onSelect={x => x} defaultValue="PICKUP" />
+      <RadioGroup defaultValue="PICKUP">
+        <Radio value="SHIP">Provide shipping address</Radio>
+        <Radio value="PICKUP">Arrange for pickup</Radio>
+      </RadioGroup>
     )
+
     expect(
       wrapper
-        .find("StyledRadio")
+        .find("Radio")
+        .first()
+        .props().selected
+    ).toBe(false)
+    expect(
+      wrapper
+        .find("Radio")
         .last()
         .props().selected
     ).toBe(true)
   })
 
-  it("renders custom radio button", () => {
+  it("selects the radio that gets clicked", () => {
+    const wrapper = mount(
+      <RadioGroup defaultValue="PICKUP">
+        <Radio value="SHIP">Provide shipping address</Radio>
+        <Radio value="PICKUP">Arrange for pickup</Radio>
+      </RadioGroup>
+    )
+
+    wrapper
+      .find("Radio")
+      .first()
+      .simulate("click")
+
+    expect(
+      wrapper
+        .find("Radio")
+        .first()
+        .props().selected
+    ).toBe(true)
+    expect(
+      wrapper
+        .find("Radio")
+        .last()
+        .props().selected
+    ).toBe(false)
+  })
+
+  it("allows the 'disabled' prop on the Radio component to take the precedence", () => {
     const wrapper = mount(
-      <RadioGroup
-        options={options}
-        onSelect={x => x}
-        renderRadio={() => (
-          <div key={Math.random()} className="customRadio">
-            found
-          </div>
-        )}
-      />
+      <RadioGroup disabled>
+        <Radio value="SHIP" disabled={false}>
+          Provide shipping address
+        </Radio>
+        <Radio value="PICKUP">Arrange for pickup</Radio>
+      </RadioGroup>
     )
-    expect(wrapper.find(".customRadio").length).toBe(2)
+
+    const radio = wrapper.find("Radio").first()
+
     expect(
       wrapper
-        .find(".customRadio")
-        .at(1)
-        .text()
-    ).toBe("found")
+        .find("Radio")
+        .first()
+        .props().disabled
+    ).toBe(false)
+    expect(
+      wrapper
+        .find("Radio")
+        .last()
+        .props().disabled
+    ).toBe(true)
   })
 
-  it("disables radio group", () => {
+  it("ignores the 'selected' prop on the Radio component", () => {
```

Comment:
@zephraph I don't think this is going to be extremely confusing but this `selected` prop would collide with the `defaultValue` prop on the parent `<RadioGroup>` component. Any thoughts?

Reply:
Yeah, that's a good point... I feel like this is indeterministic behavior. It should likely result in an error in `dev`.